spec/migrations/20200912032913_add_ancestry_to_resource_pool_spec.rb

```diff
@@ -0,0 +1,181 @@
+  let(:resource_pool_stub) { migration_stub :ResourcePool }
+  let(:resource_pool) { resource_pool_stub.create! }
+  let(:resource_pool2) { resource_pool_stub.create! }
+  let(:resource_pool3) { resource_pool_stub.create! }
+  let(:resource_pool4) { resource_pool_stub.create! }
+  let(:resource_pool5) { resource_pool_stub.create! }
+  let(:resource_pool6) { resource_pool_stub.create! }
+  let(:resource_pool7) { resource_pool_stub.create! }
+  let(:resource_pool8) { resource_pool_stub.create! }
+  let(:root) { resource_pool_stub.create! }
+
+  migration_context :up do
+    context "single parent/child rel" do
+      it 'updates ancestry' do
+        parent_rel = rel_stub.create!(:relationship => 'genealogy', :ancestry => nil, :resource_type => 'ResourcePool', :resource_id => root.id)
+        rel_stub.create!(:relationship => 'genealogy', :ancestry => parent_rel.id, :resource_type => 'ResourcePool', :resource_id => resource_pool.id)
+
+        migrate
+
+        expect(resource_pool.reload.ancestry).to eq(root.id.to_s)
+        expect(root.reload.ancestry).to eq(nil)
+        expect(rel_stub.count).to eq(0)
+      end
+    end
+
+    context "slightly more complicated tree" do
+      it 'updates ancestry' do
+        parent_rel = rel_stub.create!(:relationship => 'genealogy', :ancestry => nil, :resource_type => 'ResourcePool', :resource_id => root.id)
+        child_rel = rel_stub.create!(:relationship => 'genealogy', :ancestry => parent_rel.id, :resource_type => 'ResourcePool', :resource_id => resource_pool.id)
+        rel_stub.create!(:relationship => 'genealogy', :ancestry => child_rel.id.to_s + '/' + parent_rel.id.to_s, :resource_type => 'ResourcePool', :resource_id => resource_pool2.id)
+
+        migrate
+
+        expect(resource_pool.reload.ancestry).to eq(root.id.to_s)
+        expect(resource_pool2.reload.ancestry).to eq("#{resource_pool.id}/#{root.id}")
+        expect(root.reload.ancestry).to eq(nil)
+        expect(rel_stub.count).to eq(0)
+      end
+    end
+
+    context "complicated tree" do
+      #           a
+      #      b         c
+      #      d         g
+      #    e   f
+      it 'updates ancestry' do
+        rel_a = rel_stub.create!(:relationship => 'genealogy', :ancestry => nil, :resource_type => 'ResourcePool', :resource_id => root.id)
+        rel_c = rel_stub.create!(:relationship => 'genealogy', :ancestry => rel_a.id, :resource_type => 'ResourcePool', :resource_id => resource_pool.id)
+        rel_stub.create!(:relationship => 'genealogy', :ancestry => rel_c.id.to_s + '/' + rel_a.id.to_s, :resource_type => 'ResourcePool', :resource_id => resource_pool2.id)
+        rel_b = rel_stub.create!(:relationship => 'genealogy', :ancestry => rel_a.id.to_s, :resource_type => 'ResourcePool', :resource_id => resource_pool3.id)
+        rel_d = rel_stub.create!(:relationship => 'genealogy', :ancestry => rel_b.id.to_s + '/' + rel_a.id.to_s, :resource_type => 'ResourcePool', :resource_id => resource_pool4.id)
```

Comment:
could you introduce a helper method...

```ruby
def ancestry_string(objects)
  objects.map(&:id).join("/")
end
```

Reply:
can we stick with only commenting on one, preferably not this one, of these three, going forward please? what you actually want here is in https://github.com/ManageIQ/manageiq-schema/pull/492#discussion_r511498419